control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -668,12 +664,17 @@ func (r *HostedControlPlaneReconciler) update(ctx context.Context, hostedControl
 		return fmt.Errorf("failed to reconcile ingress operator: %w", err)
 	}
 
-	// Reconcile private IngressController
+	// Reconcile private router
 	if util.IsPrivateHCP(hostedControlPlane) {
-		r.Log.Info("Reconciling private IngressController")
+		r.Log.Info("Removing private IngressController")
+		// Ensure that if an ingress controller exists from a previous version, it is removed
```

Comment:
Am I right to say that this can be dropped after we stop supporting ocp < 4.11?
If so can we may be add a TODO with a given convention here e.g TODO: upgrade transition: drop after 4.11 support finishes? I'm concern about the growing maintenance burden / complexity we keep adding to the code base to satisfy compatible transitions, so it'd be good we keep good track of them and remove them one they stop being needed. wdyt?